src/docker/restorableappengine.cc

```diff
@@ -54,6 +71,9 @@ bool RestorableAppEngine::fetch(const App& app) {
     res = true;
   } catch (const std::exception& exc) {
     LOG_ERROR << "failed to fetch App; app: " + app.name + "; uri: " + app.uri + "; err: " + exc.what();
+    if (boost::filesystem::exists(app_dir)) {
+      boost::filesystem::remove_all(app_dir);
+    }
```

Comment:
It looks like I came across some subtle issue related to checking if app needs to be (re-)downloaded, hence I put it here as a quick fix. But, I think I will remove it from the given PR and address the issue in another one if there is indeed some issue.

Reply:
ah - something might be corrupt and we'd not re-download. seems reasonable.

Reply:
Actually, it's needed for the given feature. For example, if checkAppUpdateSize fails then App's content (App manifest, archive and manifests of images) is actually downloaded so, the follow up check isAppFetched() says yes, the App is fetched while actually it's fetched partially, so I thought that it would be OK and more correct to remove the App content (basically just manifests).
Ideally, we need to improve `isAppFetched` so it checks not just manifests but the blob store (layers) too or leave it as it is and add another method areImagesPulled() and then add it to `RestorableAppEngine::isFetched()`.

I suggest to leave this App's manifests removal and then I will improve isFetched()/isAppFetched in another PR?